src/bannerCommon/bannerActions.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect } from 'react';
+import { OphanComponentEvent } from '@guardian/libs';
+import { BrazeClickHandler } from '../utils/tracking';
+
+// Utility function which puts a try-catch block around a function and reports errors
+export const catchAndLogErrors = (description: string, fn: () => void): void => {
+    try {
+        fn();
+    } catch (e) {
+        console.log(`Error (${description}): `, e.message);
+    }
+};
+
+// Pass a function to run when the user hits the escape key
+// Useful for enabling a keyboard shortcut for dismissing banners
+export function useEscapeShortcut(
+    eventHandler: (event: KeyboardEvent) => void,
+    deps: React.DependencyList = [],
+): void {
+    function handleEscapeKeydown(event: KeyboardEvent) {
+        // IE key name is 'Esc', because IE
+        const isEscapeKey = event.key === 'Escape' || event.key === 'Esc';
+        if (isEscapeKey) {
+            eventHandler(event);
+        }
+    }
+
+    useEffect(() => {
+        window.addEventListener('keydown', handleEscapeKeydown);
+
+        return () => window.removeEventListener('keydown', handleEscapeKeydown);
+    }, deps);
+}
+
+export type OnCloseClick = (
+    evt: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    internalButtonId: number,
+) => void;
+
+// Function to log banner close events
+export const logBannerCloseToOphan = (
```

Comment:
I _think_ maybe this is more generic than closing the banner, in that it can be used to log _any_ banner button click (i.e. there's nothing specific to closing banners in here I don't think). What do you think of renaming to reflect that? Also relates to comment above about the name here making is appear that this is only related to Ophan when in fact it also logs clicks to Braze.

Reply:
See comment above re discovering code already exists in `utils/tracking.ts` file, and changing to use that code in PR Part 4